src/vaults/AsyncRequests.sol

```diff
@@ -216,6 +218,40 @@ contract AsyncRequests is BaseInvestmentManager, IAsyncRequests {
     }
 
     // -- Gateway handlers --
+    /// @inheritdoc IInvestmentManagerGatewayHandler
+    function approvedDeposits(PoolId poolId, ShareClassId scId, AssetId assetId, uint128 assetAmount, D18 pricePoolPerAsset) external auth {
+        (address asset, uint256 tokenId) = poolManager.idToAsset(assetId);
+
+        balanceSheet.noteDeposit(poolId, scId, asset, tokenId, address(globalEscrow()), assetAmount, pricePoolPerAsset);
+        address poolEscrow = address(poolEscrowProvider.escrow(poolId.raw()));
+        globalEscrow().authTransferTo(asset, tokenId, poolEscrow, assetAmount);
+    }
+
+    /// @inheritdoc IInvestmentManagerGatewayHandler
+    function issuedShares(PoolId poolId, ShareClassId scId, uint128 shareAmount, D18 pricePoolPerShare) external auth {
+        balanceSheet.issue(poolId, scId, globalEscrow(), pricePoolPerShare, shareAmount);
+    }
+
+    /// @inheritdoc IInvestmentManagerGatewayHandler
+    function revokedShares(
+        PoolId poolId,
+        ShareClassId scId,
+        AssetId assetId,
+        uint128 assetAmount,
+        uint128 shareAmount,
+        D18 pricePoolPerShare
+    ) external auth {
+        (address asset, uint256 tokenId) = poolManager.idToAsset(assetId);
+        // Lock assets to ensure they are not withdrawn and are available for the redeeming user
+        poolEscrowProvider.escrow(poolId.raw()).reserveIncrease(scId.raw(), asset, tokenId, assetAmount);
+
+        IAsyncVault vault_ = IAsyncVault(vault[poolId.raw()][scId.raw()][assetId.raw()]);
+
+        // Need to transfer to the balanceSheet so that it can burn from itself
+        globalEscrow().authTransferTo(vault_.share(), address(balanceSheet), shareAmount);
+        balanceSheet.revoke(poolId, scId, address(globalEscrow()), pricePoolPerShare, shareAmount);
```

Comment:
Based on the comment
> Need to transfer to the balanceSheet so that it can burn from itself

Shouldn't the `from` be the BS here? Otherwise the BS burns from the provided global escrow

```suggestion
        balanceSheet.revoke(poolId, scId, address(balanceSheet), pricePoolPerShare, shareAmount);
```

Reply:
Yes. Good catch!!

Reply:
Fixed in https://github.com/centrifuge/protocol-v3/pull/306/commits/fb99eb92ee7417fa60e6f7bc29fa5ddc33f170bd